snyk/client.py

```diff
@@ -76,25 +104,65 @@ def put(self, path: str, body: Any) -> requests.Response:
             raise SnykHTTPError(resp)
         return resp
 
-    def get(self, path: str) -> requests.Response:
-        url = "%s/%s" % (self.api_url, path)
-        logger.debug("GET: %s" % url)
+    def get(
+        self, path: str, params: dict = None, version: str = None
+    ) -> requests.Response:
+        """
+        V3 Compatible Snyk Client, assumes the presence of Version, either set in the client
+        or called in this method means that we're talking to a V3 endpoint and will ensure the
+        params are encoded properly with the version.
+
+        Since certain endpoints can exist only in certain versions, being able to override the
+        client version with each GET is necessary
+
+        Returns a standard requests Response object
+        """
+
+        path = cleanup_path(path, logger)
+
+        url = f"{self.api_url}/{path}"
+
+        if params or self.version:
+
+            if not params:
+                params = {}
+
+            # we use the presence of version to determine if we are v3 or not
+            if "version" not in params.keys() and self.version:
+                params["version"] = version or self.version
+
+            # Python Bools are True/False, JS Bools are true/false
+            # Snyk v3 API is strictly case sensitive at the moment
+
+            for k, v in params.items():
+                if isinstance(v, bool):
+                    params[k] = str(v).lower()
+
+            debug_url = f"{url}&{urllib.parse.urlencode(params)}"
+            fkwargs = {"headers": self.api_headers, "params": params}
+        else:
+            debug_url = url
+            fkwargs = {"headers": self.api_headers}
```

Comment:
Given we're adding the params key, could we make that work with the V1 API as well at the same time, in the same way as above. It could you provide a default value for the headers?

Reply:
The logic will include params in a GET to any url, but there's no v1 endpoint that uses params + GET. Params are only used in the context of POSTs.